xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/ModelFactory.java

```diff
@@ -1063,4 +1076,40 @@ public JobLog toRestJobLog(Collection<LogEvent> logs, URI self)
 
         return log;
     }
+
+    /**
+     * check if the given property should be exposed via REST
+     * @param restProperty the property to be read/written
+     * @param forWrite true iff we want to write to the XObject property
+     * @return true if the property is considered public for read or write
+     */
+    private boolean isTypePublic(Property restProperty, boolean forWrite)
+    {
+        if (PASSWORD_TYPE.equals(restProperty.getType())) {
+            return false;
+        }
+
+        if (!forWrite && EMAIL_TYPE.equals(restProperty.getType())) {
```

Comment:
I have to admit that this part looks a bit like overkill; if this is not wanted I can roll back the last commit or remove this code from the PR.

Reply:
I thought we were already obfuscating the email in the REST API. This is not the case? If not, then yes it should be done. Would be better to have it in a different JIRA issue and different PR though.

Reply:
I have seen the email in plain when testing.
The corresponding issue would be XWIKI-8331 then.